balancer-js/examples/pools/composable-stable/createAndInitJoin.ts

```diff
@@ -1,141 +1,215 @@
     tx.hash
   );
-  console.log('receipt: ' + receipt);
 
   const composableStableFactoryInterface = new Interface(
     composableStableFactoryAbi
   );
 
+  const composableStablePoolInterface = new Interface(composableStablePoolAbi);
+
   const poolCreationEvent: LogDescription | null | undefined = receipt.logs
-    .filter((log: Log) => log.address === contractAddress)
+    .filter((log: Log) => {
+      return log.address.toUpperCase() === contractAddress.toUpperCase();
+    })
     .map((log) => {
       try {
-        return composableStableFactoryInterface.parseLog(log);
-      } catch {
+        const parsedLog = composableStableFactoryInterface.parseLog(log);
+        return parsedLog;
+      } catch (error) {
+        console.error(error);
         return null;
       }
     })
     .find((parsedLog) => parsedLog?.name === 'PoolCreated');
+
   if (!poolCreationEvent) return console.error("There's no event");
-  const poolAddress: string = poolCreationEvent.args.pool;
 
-  const pool: PoolWithMethods | undefined= await balancer.pools.findBy(
-    'address',
-    poolAddress
+  const poolAddress: string = poolCreationEvent.args.pool;
+  const pool = new Contract(
+    poolAddress,
+    composableStablePoolInterface,
+    provider
   );
+  const poolId = await pool.getPoolId();
 
-  console.log('Pool Id: ' + pool?.id);
-
-  if(!pool) return console.error("No pool was found by balancer.pools.find functionality");
+  const networkConfig = getNetworkConfig(sdkConfig);
 
   const {
-    to: initJoinTo,
-    data: initJoinData,
-    attributes,
-  } = pool.buildInitJoin(await signer.getAddress(), tokenAddresses, [
-    '100',
-    '100',
-  ]);
+    tokens: { wrappedNativeAsset },
+  } = networkAddresses(networkConfig.chainId);
+
+  const iERC20 = [
+    'function approve(address,uint256) nonpayable',
+    'function balanceOf(address) view returns(uint)',
+  ];
+  const tokenBalances = (
+    await getBalances([...tokenAddresses, poolAddress], signer, signerAddress)
+  ).map((b) => b.toString());
+
+  console.log('tokenBalances');
+  console.log(tokenBalances);
+
+  const initJoinParams = composableStablePoolFactory.buildInitJoin({
+    joiner: signerAddress,
+    poolId,
+    poolAddress,
+    tokensIn: [...tokenAddresses, poolAddress],
+    amountsIn: [
+      scale(new BigNumber('1'), 18).toString(),
+      scale(new BigNumber('1'), 18).toString(),
+      '0',
+    ],
+    wrappedNativeAsset,
+  });
 
-  console.log('initJoin target address: ' + initJoinTo);
-  console.log('initJoin data: ' + initJoinData);
-  console.log('initJoin attributes: ' + attributes);
+  const erc20 = new Contract(AddressZero, iERC20);
 
+  // Approve vault for seeder
+  await Promise.all(
+    Object.values(tokenAddresses).map((address) => {
+      return erc20
+        .attach(address)
+        .connect(signer)
+        .approve(to, parseEther('100').toString());
+    })
+  );
+
+  const initJoinTx = await signer.sendTransaction({
```

Comment:
I'm receiving Error BAL#506 JOIN_ABOVE_MAX, like the user doesn't have enough tokens to fulfill the transaction.